test/integration/targets/aws_ses_identity_policy/tasks/main.yaml

```diff
@@ -0,0 +1,268 @@
+
+  always:
+    - name: clean-up identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: absent
+        <<: *aws_connection_info
+# ============================================================
+- name: test add identity policy by identity arn
+  block:
+    - name: register identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: present
+        <<: *aws_connection_info
+      register: identity_info
+
+    - name: register identity policy
+      aws_ses_identity_policy:
+        identity: "{{ identity_info.identity_arn }}"
+        policy_name: "{{ policy_name }}"
+        policy: "{{ lookup('template', 'policy.json.j2') }}"
+        state: present
+        <<: *aws_connection_info
+      register: result
+
+    - name: assert result.changed == True
+      assert:
+        that:
+          - result.changed == True
+
+    - name: assert result.policies contains only policy
+      assert:
+        that:
+          - result.policies|length == 1
+          - result.policies|select('equalto', policy_name)|list|length == 1
+
+  always:
+    - name: clean-up identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: absent
+        <<: *aws_connection_info
+# ============================================================
+- name: test add multiple identity policies
+  block:
+    - name: register identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: present
+        <<: *aws_connection_info
+      register: identity_info
+
+    - name: register identity policy
+      aws_ses_identity_policy:
+        identity: "{{ domain_identity }}"
+        policy_name: "{{ policy_name }}-{{ item }}"
+        policy: "{{ lookup('template', 'policy.json.j2') }}"
+        state: present
+        <<: *aws_connection_info
+      with_items:
+        - 1
+        - 2
+      register: result
+
+    - name: assert result.policies contains policies
+      assert:
+        that:
+          - result.results[1].policies|length == 2
+          - result.results[1].policies|select('equalto', policy_name + '-1')|list|length == 1
+          - result.results[1].policies|select('equalto', policy_name + '-2')|list|length == 1
+
+  always:
+    - name: clean-up identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: absent
+        <<: *aws_connection_info
+# ============================================================
+- name: test remove identity policy
+  block:
+    - name: register identity
+      aws_ses_identity:
+        identity: "{{ domain_identity }}"
+        state: present
+        <<: *aws_connection_info
+      register: identity_info
+
+    - name: register identity policy
+      aws_ses_identity_policy:
+        identity: "{{ domain_identity }}"
+        policy_name: "{{ policy_name }}"
+        policy: "{{ lookup('template', 'policy.json.j2') }}"
```

Comment:
In addition to the `lookup` method of taking in the policy, can you also try a policy argument like: 

```
policy:
  Id: SampleAuthorizationPolicy
  Version: "2012-10-17"
  Statement:
  - Sid: DenyAll
    Effect: Deny
    Resource: "{{ identity_info.identity_arn }}"
    Principal: "*"
    Action: "*"
```

The JSON argument type should automatically take that as well, but I'd like to have that validated in the module tests to enforce that UI.

Reply:
That's cool, I didn't even realize that was possible.

New test added, but I've put it further up the file with the rest of the register variants.

Reply:
Thank you!